service_container/service_closures.rst

```diff
@@ -0,0 +1,112 @@
+.. index::
+    single: DependencyInjection; Service Closures
+
+Service Closures
+================
+
+This feature wraps the injected service into a closure allowing it to be
+lazily loaded when and if needed.
+This is useful if the service being injected is a bit heavy to instantiate
+or is used only in certain cases.
+The service is instantiated the first time the closure is called, while
+all subsequent calls return the same instance, unless the service is
+:doc:`not shared </service_container/shared>`::
+
+    // src/Service/MyService.php
+    namespace App\Service;
+
+    use Symfony\Component\Mailer\MailerInterface;
+
+    class MyService
+    {
+        /**
+         * @var callable(): MailerInterface
+         */
+        private \Closure $mailer;
+
+        public function __construct(\Closure $mailer)
+        {
+            $this->mailer = $mailer;
+        }
+
+        public function doSomething(): void
+        {
+            // ...
+
+            $this->getMailer()->send($email);
+        }
+
+        private function getMailer(): MailerInterface
+        {
+            return ($this->mailer)();
```

Comment:
Do we need the getMailer() method or can we interact with the property directly?

Reply:
No, with the improved phpdoc, this becomes useless, removed.

Reply:
I would have kept it, the phpdocs helps understanding the code when crossing the private closure, but this method ensures the expected type is returned at runtime.

Reply:
I'm OK either way, so when you come to an agreement, please let me know and I'll update the PRs.

Reply:
Hi guys, I was wondering if you've reached any consensus regarding this one?

Reply:
Lets go with @HeahDude's comment then, easier way to understand is always a plus for me

Reply:
Done